src/iiif/ingress_zip_consumer.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+import os
+import shutil
+
+from django.conf import settings
+from ingress.consumer.base import BaseConsumer
+
+from iiif import tools
+from main import settings
+
+logger = logging.getLogger(__name__)
+
+
+class ZipConsumer(BaseConsumer):
+    collection_name = settings.ZIP_COLLECTION_NAME
+
+    """
+    Whether or not to immediately remove messages once consumption succeeds.
+    If set to False, message.consume_succeeded_at will be set.
+    """
+    remove_message_on_consumed = False
+
+    """
+    Whether or not to set Message.consume_started_at immediately once consumption starts
+    """
+    set_consume_started_at = True
+
+    def consume_raw_data(self, raw_data):
+        record = json.loads(raw_data)
+
+        # Create a tmp folder to store downloaded source files
+        zipjob_uuid, tmp_folder_path = tools.create_tmp_folder()
+
+        # Init contents of txt info file which is send along in the zip
```

Comment:
This comment is confusing. This loop actually moves the files into a single folder to be zipped afterwards. Alongside that it keeps a log of files added or not. But this comment made me think this just creates a txt file with info, which is not the case.